nltk/tokenize/casual.py

```diff
@@ -0,0 +1,304 @@
+        [\(]?
+        \d{3}
+        [\-\s.\)]*
+      )?
+      \d{3}          # exchange
+      [\-\s.]*
+      \d{4}          # base
+    )"""
+    ,
+    # ASCII Emoticons
+    EMOTICONS
+    ,
+    # HTML tags:
+    r"""<[^>\s]+>"""
+    ,
+    # ASCII Arrows
+    r"""[\-]+>|<[\-]+"""
+    ,
+    # Twitter username:
+    r"""(?:@[\w_]+)"""
+    ,
+    # Twitter hashtags:
+    r"""(?:\#+[\w_]+[\w\'_\-]*[\w_]+)"""
+    ,
+
+    # Remaining word types:
+    r"""
+    (?:[a-z][a-z'\-_]+[a-z])       # Words with apostrophes or dashes.
+    |
+    (?:[+\-]?\d+[,/.:-]\d+[+\-]?)  # Numbers, including fractions, decimals.
+    |
+    (?:[\w_]+)                     # Words without apostrophes or dashes.
+    |
+    (?:\.(?:\s*\.){1,})            # Ellipsis dots.
+    |
+    (?:\S)                         # Everything else that isn't whitespace.
+    """
+    )
+
+######################################################################
+# This is the core tokenizing regex:
+
+WORD_RE = re.compile(r"""(%s)""" % "|".join(REGEXPS), re.VERBOSE | re.I
+                     | re.UNICODE)
+
+# The emoticon string gets its own regex so that we can preserve case for
+# them as needed:
+EMOTICON_RE = re.compile(EMOTICONS, re.VERBOSE | re.I | re.UNICODE)
+
+# These are for regularizing HTML entities to Unicode:
+HTML_ENTITY_DIGIT_RE = re.compile(r"&#\d+;")
+HTML_ENTITY_ALPHA_RE = re.compile(r"&\w+;")
+AMP = "&amp;"
+
+######################################################################
+
+class TweetTokenizer:
+    """Tokenize Tweets"""
+    def __init__(self, preserve_case=True):
+        self.preserve_case = preserve_case
+
+    def tokenize(self, text):
+        """
+        :param s: str
+        :rtype: list(str)
+        :return: a tokenized list of strings; concatenating this list returns
+        the original string if preserve_case=False
+        """
+        # Try to ensure unicode:
+        try:
+            text = str(text)
+        except UnicodeDecodeError:
+            text = str(text).encode('string_escape')
+            text = str(text)
+        # Fix HTML character entities:
+        text = self._html2unicode(text)
+        # Tokenize:
+        words = WORD_RE.findall(text)
+        # Possibly alter the case, but avoid changing emoticons like :D into :d:
+        if not self.preserve_case:
+            words = list(map((lambda x : x if EMOTICON_RE.search(x) else
+                              x.lower()), words))
+        return words
+
+
+    def _html2unicode(self, text):
+        """
+        Try to replace all the HTML entities in `s` with their corresponding
+        Unicode characters.
+        """
+        # First the digits:
+        ents = set(HTML_ENTITY_DIGIT_RE.findall(text))
+        if len(ents) > 0:
+            for ent in ents:
+                entnum = ent[2:-1]
+                try:
+                    entnum = int(entnum)
+                    text = text.replace(ent, chr(entnum))
+                except:
```

Comment:
I think `except Exception` goes against the spirit of the PEP8 recommendation!

@kmike I don't have strong feelings about how to address this; both your suggestions seem reasonable. Given that this seems to be an area where you have experience, would you consider fixing up `casual.py` along the lines you recommend?